### ISSUE TITLE
Native driver not found when using Python 3.12

### ISSUE THREAD

**Issue by rjoelnorgren** [ClickHouse/dbt-clickhouse#264]
### Describe the bug
Unable to run dbt commands that connect to the warehouse when using the native driver, python 3.12, and dbt-core 3.7.x, and dbt-clickhouse 1.7.x.
```
dbt@e56ea02a991d:/opt/dbt$ dbt seed
21:10:01  Running with dbt=1.7.11
21:10:01  Registered adapter: clickhouse=1.7.4
21:10:01  Found 143 models, 3 snapshots, 13 seeds, 1 operation, 166 tests, 68 sources, 0 exposures, 0 metrics, 590 macros, 0 groups, 0 semantic models
21:10:01  
21:10:01  
21:10:01  Finished running  in 0 hours 0 minutes and 0.07 seconds (0.07s).
21:10:01  Encountered an error:
Database Error
  Database Error
    Native adapter required but package clickhouse-driver is not installed
```

### Steps to reproduce
1. Install dbt-clickhouse==1.7.4, dbt-core==1.7.11, clickhouse-driver==0.2.7 with Python 3.12.2
2. Run a command that interacts with database

### Expected behaviour
The command runs

#### Environment
* dbt version: 1.7.x
* dbt-clickhouse version: 1.7.x
* clickhouse-driver version: 0.2.7
* clickhouse-connect version (if using http):
* Python version: 3.12.2
* Operating system: Debian (bookworm)

#### ClickHouse server
* ClickHouse Server version: 24.3

**Comment by genzgd** [comment]
can you validate that clickhouse-driver successfully installed in your dbt environment?  Something like

```
python -c "import clickhouse_driver"
```
should not return anything if it's correctly installed.

**Comment by rjoelnorgren** [comment]
@genzgd
```
dbt@a954ce9e82c1:/opt/dbt$ python -c "import clickhouse_driver"
dbt@a954ce9e82c1:/opt/dbt$ 
```

We also use clickhouse-driver in the same container to run non-dbt workloads and they are working in 3.12.

**Comment by genzgd** [comment]
Okay, I ran into clickhouse_driver issues in both a docker 3.12 container and a local 3.12 virtual environment that seemed to be this [issue](https://stackoverflow.com/questions/77364550/attributeerror-module-pkgutil-has-no-attribute-impimporter-did-you-mean).   It seemed to work after I upgraded setuptools, so we've included a recent version of setuptools in the setup.py.

Native tests now pass in Github actions and Python 3.12, so please try the new release 1.7.5, or try upgrading setuptools in your Python 3.12 environment.

**Comment by rjoelnorgren** [comment]
@genzgd, confirming that the issue is resolved after upgrading to 1.7.5. Thank you so much!
